File: info/1. Алгоритмы/2. Сортировка/4. Сортировка пузырьком/bubble_sort/bubble_sort.py

list_1 = [5, 0, -2, 7, 3]

sorted_index = len(list_1)
while True:
    number_of_swap = 0
    for i in range(0, sorted_index-1):
        if list_1[i] > list_1[i+1]:
            list_1[i], list_1[i+1] = list_1[i+1], list_1[i]
            number_of_swap += 1
    sorted_index -= 1
    if number_of_swap == 0:
        break

print(list_1)

File: info/1. Алгоритмы/2. Сортировка/4. Сортировка пузырьком/bubble_sort/BubbleSort.java

import java.util.Arrays;

public class BubbleSort {

    public static void main(String[] args) {

        int[] array = new int[] { 5, 0, -2, 7, 3 };

        int sortedIndex = array.length;
        int numberOfSwap = 1;
        while (numberOfSwap > 0) {
            numberOfSwap = 0;
            for (int i = 0; i < sortedIndex - 1; i++) {
                if (array[i] > array[i + 1]) {
                    int temp = array[i];
                    array[i] = array[i + 1];
                    array[i + 1] = temp;
                    numberOfSwap += 1;
                }
            }
            sortedIndex -= 1;

        }
        System.out.println(Arrays.toString(array));

    }

}
